pkg/host/store.go

```diff
@@ -129,6 +130,11 @@ func (s *Store[E]) Update(ctx context.Context, obj E) (E, error) {
 		return obj, nil
 	}
 
+	if oldObj.GetGeneration() != obj.GetGeneration() {
```

Comment:
I believe it would be better to perform this check before deepEqual. If the check fails, it indicates that the file in the store has already been modified, and no further operations with this object are necessary for the current reconciliation.

Reply:
Done